### PR TITLE
CDP: add CDPMsgPowerRequest and CDPMsgPowerAvailable

### PR DESCRIPTION
**Pull request by cq674350529**
refine CDPMsgPowerRequest and CDPMsgPowerAvailable based on the output from Whireshark

For your convenience, a sample of pcap is attached

[cdp_power.zip](https://github.com/secdev/scapy/files/10124456/cdp_power.zip)

### CODE REVIEW

**Comment by mspncp** [comment]
fly-by-comment: there is a typo in the subject line of your commit (and the PR title):   CDPMsgPowerAavaiable ->  CDPMsgPowerAavailable

**Comment by cq674350529** [comment]
@mspncp Thanks, fixed the typos :)

File: scapy/contrib/cdp.py [hunk 1]
```diff
@@ -352,9 +354,26 @@ class CDPMsgMgmtAddr(CDPMsgAddr):
...
-    type = 0x0019
+class CDPMsgPowerRequest(CDPMsgGeneric):
+    name = "Power Request"
+    fields_desc = [XShortEnumField("type", 0x0019, _cdp_tlv_types),
+                   FieldLenField("len", None, "power_requested_list", fmt="!H",
+                                 adjust=lambda pkt, x: x + 8),
+                   ShortField("req_id", 0),
+                   ShortField("mgmt_id", 0),
+                   FieldListField("power_requested_list", [], IntField("power_requested", 0),
+                                  count_from=lambda pkt: (pkt.len - 8) / 4)]
```
Comment (gpotter2) [inline scapy/contrib/cdp.py:365]:
```suggestion
                                  count_from=lambda pkt: (pkt.len - 8) // 4)]
```

File: scapy/contrib/cdp.py [hunk 2]
```diff
@@ -352,9 +354,26 @@ class CDPMsgMgmtAddr(CDPMsgAddr):
...
+
+class CDPMsgPowerAvailable(CDPMsgGeneric):
+    name = "Power Available"
+    fields_desc = [XShortEnumField("type", 0x001a, _cdp_tlv_types),
+                   FieldLenField("len", None, "power_available_list", fmt="!H",
+                                 adjust=lambda pkt, x: x + 8),
+                   ShortField("req_id", 0),
+                   ShortField("mgmt_id", 0),
+                   FieldListField("power_available_list", [], IntField("power_available", 0),
+                                  count_from=lambda pkt: (pkt.len - 8) / 4)]
```
Comment (gpotter2) [inline scapy/contrib/cdp.py:376]:
```suggestion
                                  count_from=lambda pkt: (pkt.len - 8) // 4)]
```

**Review by gpotter2 — CHANGES_REQUESTED** [review]
Hi ! really sorry for the delay. Could you add a unit test to https://github.com/secdev/scapy/blob/master/test/contrib/cdp.uts ? Thanks !

**Comment by cq674350529** [comment]
Thanks for the suggestions, I updated them and added an unit test for it.

**Review by gpotter2 — APPROVED** [review]

**Comment by gpotter2** [comment]
Very sorry for the delay !

**Pull request merged**